src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/update/UpdateClause.java

```diff
@@ -37,29 +39,54 @@ public record UpdateClause(EnumMap<UpdateOperator, ObjectNode> updateOperationDe
    */
   public List<UpdateOperation> buildOperations() {
     // First, resolve operations individually; this will handle basic validation
-    var operationMap = new EnumMap<UpdateOperator, UpdateOperation>(UpdateOperator.class);
+    final var operationMap = new EnumMap<UpdateOperator, UpdateOperation>(UpdateOperator.class);
     updateOperationDefs
         .entrySet()
         .forEach(e -> operationMap.put(e.getKey(), e.getKey().resolveOperation(e.getValue())));
 
-    // Then handle cross-operation validation
-
-    // First: verify $set and $unset do NOT have overlapping keys
-
-    UpdateOperation<?> setOp = operationMap.get(UpdateOperator.SET);
-    UpdateOperation<?> unsetOp = operationMap.get(UpdateOperator.UNSET);
+    // Then handle cross-operation validation: first, exact path conflicts:
+    final var actionMap = new TreeMap<PathMatchLocator, UpdateOperator>();
+    operationMap
+        .entrySet()
+        .forEach(
+            e -> {
+              final UpdateOperator type = e.getKey();
+              List<ActionWithLocator> actions = e.getValue().actions();
+              for (ActionWithLocator action : actions) {
+                UpdateOperator prevType = actionMap.put(action.locator(), type);
+                if (prevType != null) {
+                  throw new JsonApiException(
+                      ErrorCode.UNSUPPORTED_UPDATE_OPERATION_PARAM,
+                      "Update operators '%s' and '%s' must not refer to same path: '%s'"
+                          .formatted(prevType.operator(), type.operator(), action.locator()));
+                }
+              }
+            });
 
-    if ((setOp != null) && (unsetOp != null)) {
-      Set<String> paths = getPaths(setOp);
-      paths.retainAll(getPaths(unsetOp));
+    // Second: check for sub-paths -- efficiently done by using alphabetic-sorted paths
+    // and comparing adjacent ones; for each pair see if later one is longer one and
+    // starts with dot.
+    // For example: "path.x" vs "path.x.some.more" we notice latter has suffix ".some.more"
```

Comment:
I don't remember off the top of my head which characters we allow in field names, but I don't think this will work if we allow spaces, `$`, or `+` in field names. Because "path.x$foo" comes before "path.x.foo" in lexicographical order.

Is there any way we can use a different approach here? I'd rather not add code that blocks us from supporting "$" in field names.

Reply:
@tatu-at-datastax the issue is your alphabetical sorting assumption. The assumption here is that `path.x.some.more` would _immediately_ follow `path.x` in the sort order, unless there's some other conflicting path. But in ASCII, `$` is less than `.`. So conventional string sorting would put fields with dollar paths _before_ conflicting paths. Below is an example in JavaScript

```
> ['path.x', 'path.x.some.more', 'path.x$foo'].sort()
[ 'path.x', 'path.x$foo', 'path.x.some.more' ]
```

Not sure if that applies here, just figured I'd call that out.

Reply:
@vkarpov15 Ah. Ok, let me think this over. Good catch.

Reply:
I can make sorting work with some more work. Two obvious extensions:

1. Use modified String comparison where `.` is sorted like `null` byte (value 0)
2. Sort split segments (which Locator already has)

both would work; I may go with (2)

Reply:
Added segment-aware sorting, testing to verify functioning.

Reply:
Nice, thanks. It looks like the reason why this works though is because you've included a loop over path segments in the `compare()` method - any worries about this being potentially slow? You'll have to iterate over path segments O(nlogn) times, creating a map of all parent paths will likely be faster but more memory intensive. I don't think this is a big deal either way, just food for thought.

Reply:
@vkarpov15 that's a reasonable question. I guess it all depends on depth of document. My thinking was that for deeper documents most differences are at first segment(s) anyway so that it evens out, compared to String comparison. But my main consideration was, I guess, between straight String comparison (just changing sort value of `.`) vs segmented, where I think performance is similar. I did not consider possibility of building lookup for covered paths: in that case it does come back to both memory usage added and (more importantly), construction of new String instances to reconstruct parent paths. I guess it can go either way wrt overall performance, in the end.

Fundamentally I guess it also comes down to how many updates do we usually have: I suspect differences are not meaningful until into thousands of operations.

Since this logic is fairly isolated, I think I'll keep the question in mind and can reconsider re-implementation if we find a hotspot.